src/plugins/home/server/services/sample_data/sample_data_registry.ts

```diff
@@ -52,6 +58,12 @@ export class SampleDataRegistry {
     );
     createUninstallRoute(router, this.sampleDatasets, usageTracker);
 
+    if (customIntegrations) {
+      this.sampleDatasets.forEach((dataSet) => {
+        registerSampleDataSetWithIntegration(customIntegrations, core, dataSet);
+      });
+    }
+
     return {
       registerSampleDataset: (specProvider: SampleDatasetProvider) => {
```

Comment:
`registerSampleDataset` currently isn't used anywhere in Kibana... I'd vote to move this functionality to a private method and remove it from the public contract altogether. (Or if you don't want to do it in this PR, Core can make a follow-up one to clean it up).

Probably this would mean calling `this.registerSampleDataset` as a first step to actually register the default `sampleDatasets` (this would have the added benefit of doing a runtime validation on the sample data which could help prevent any regressions)

Reply:
Removed from public API, and made it private method. Move initialization of sample data-sets into the setup.

Reply:
cc @kobelb @stacey-gammon for visibility

Reply:
@thomasneirynck Would you mind also updating the `README` to remove mention of this method?

API docs will need to be regenerated too since it is in the new built docs (see `/api_docs/home.json`)